### PR TITLE
Update SDK Tracing spec in regards to shared communication objects.

### PR DESCRIPTION
**Pull request by z1c0**

### CODE REVIEW

File: specification/sdk-tracing.md [hunk 1]
```diff
@@ -110,8 +110,18 @@ invocations. The span processors are invoked only when
...
+invoked in the same order as they were registered.
+
+All `Tracer` instances created by a `TracerFactory` share the same span processors.
+Changes to this collection reflect in all `Tracer` instances.
+Implementation-wise, this could mean that `Tracer` instances have a reference to
+their `TracerFactory` and can access span processor objects only via this
+reference.
+
+Manipulation of the span processors collection must only happen on `TracerFactory`
+instances. This means methods like `addSpanProcessor` must move from `Tracer`
```
Comment (Oberon00) [inline specification/sdk-tracing.md]: Phrases like "Must move" do not belong in the spec. The spec describes how the languages should *be*, not how they need to change from some prior version.

Comment (z1c0) [reply specification/sdk-tracing.md]: Sure. You could have just corrected that without the lecturing part, though ;-)

**Comment by z1c0** [comment]
> I'd add a note somewhere that this also affects exporters and any other SDK-specific (or vendor API implementation specific) configuration

* The first paragraph on span processors already covers exporters.
> This interface must be used to implement span exporter to batch and convert spans.

* Propagators are missing entirely in the SDK spec. This is worth a separate PR.

* *Other* configuration objects: this could go into the "Tracer Creation" topic. Agreed?

File: specification/sdk-tracing.md [hunk 2]
```diff
@@ -110,8 +116,18 @@ invocations. The span processors are invoked only when
 must be used to implement [span exporter](#span-exporter) to batch and convert
 spans.
 
-Span processors can be registered directly on SDK Tracer and they are invoked in
-the same order as they were registered.
+Span processors can be registered directly on SDK `TracerFactory` and they are
+invoked in the same order as they were registered.
+
+All `Tracer` instances created by a `TracerFactory` share the same span processors.
+Changes to this collection reflect in all `Tracer` instances.
```
Comment (Flarna) [inline specification/sdk-tracing.md]:
```suggestion
Changes to this collection reflects in all `Tracer` instances.
```

File: specification/sdk-tracing.md [hunk 3]
```diff
@@ -110,8 +116,18 @@ invocations. The span processors are invoked only when
...
+Implementation-wise, this could mean that `Tracer` instances have a reference to
+their `TracerFactory` and can access span processor objects only via this
```
Comment (Flarna) [inline specification/sdk-tracing.md]: Above it's mentioned that `Tracers` are created via the `TraceFactory` but reading this here sounds like we could have more then one `TraceFactory`. If there are more factories, how to get that one I should use in instrumentation code to create a tracer?

File: specification/sdk-tracing.md (hunk 2 above)
Comment (Oberon00) [reply specification/sdk-tracing.md]: It's "change**s**", so reflect without "s" seems correct.

File: specification/sdk-tracing.md (hunk 3 above)
Comment (z1c0) [reply specification/sdk-tracing.md]: By default there is a global instance of `TracerFactory` available, but there could be more than one `TracerFactory`.
It's basically the same way as it was for `Tracer` instances before. There was a global instance but you could always `new Tracer`.

Comment (Flarna) [reply specification/sdk-tracing.md]: So we may get a `TracerFactoryFactory`? Hopefully that one will never have more then one instance... 😁

Comment (z1c0) [reply specification/sdk-tracing.md]: On an API level, there will be the global "TracerFactory" (the comment above was meant for SDKs)

**Comment by z1c0** [comment]
I'd like to move this forward and have a PR for the Otel spec repo ready.
Please approve (or comment).

**Pull request merged**